src/backend/distributed/planner/relation_restriction_equivalence.c

```diff
@@ -1885,6 +1885,12 @@ GetRestrictInfoListForRelation(RangeTblEntry *rangeTblEntry,
 			continue;
 		}
 
+		/* Currently we don't know how to process Subplans */
+		if (IsA(restrictionClause, SubPlan))
```

Comment:
Here we could also put the condition on when we want to continue to also cover other possible failures. I will need to look more into what other types this can or cannot be.